### PR TITLE
Report applied Kubernetes manifests for the ScriptWrapper-based Octopus Kubernetes steps

### PR DESCRIPTION
**Pull request by kevjt**
Adds manifest reporting capability (via service messages) to the Octopus Kubernetes steps that are implemented using the ScriptWrapper. These steps are currently:

- Configure and apply a Kubernetes Ingress
- Configure and apply a Kubernetes Service

The following steps are script wrapper steps by do not yet support manifest reporting as of this PR. The reason is due to the way that secrets and configmaps are currently being created via `kubectl create --from-file` in the script, instead of via a `kubectl apply`. This means that there isn't a yaml manifest available for the script wrapper to report on ahead of the actual execution. 
- Configure and apply a Kubernetes ConfigMap
- Configure and apply a Kubernetes Secret
- Configure and apply Kubernetes resources (Deployment, Ingress, and Service objects get reported but secret and configmap objects do not for the same reason above) 

Config maps and secrets will be handled separately in a future PR, as they require additional Octopus Server changes.

[sc-98555]

### CODE REVIEW

File: source/Calamari.Tests/KubernetesFixtures/ManifestReportScriptWrapperTests.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            wrapper.IsEnabled(Syntax).Should().BeFalse();
+            manifestReporter.DidNotReceive().ReportManifestApplied(Arg.Any<string>());
+        }
+        
+        [TestCase(SpecialVariables.ClusterUrl, "MyClusterUrl")]
+        [TestCase(MachineVariables.DeploymentTargetType, "KubernetesTentacle")]
+        [TestCase(SpecialVariables.AksClusterName, "Aks")]
+        [TestCase(SpecialVariables.EksClusterName, "Eks")]
+        [TestCase(SpecialVariables.GkeClusterName, "Gke")]
+        public void Enabled_WhenDeployingToAKubernetesCluster(string variableName, string variableValue)
```
[the remark this answers is not in the document]
Comment (kevjt) [reply source/Calamari.Tests/KubernetesFixtures/ManifestReportScriptWrapperTests.cs:48]: The isEnabled check just calls [IsKubernetesScript](https://github.com/OctopusDeploy/Calamari/blob/main/source/Calamari/Kubernetes/KubernetesVariableExtensions.cs#L8). Those variable names are what that helper currently looks for

File: source/Calamari.Tests/KubernetesFixtures/ManifestReportScriptWrapperTests.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            var variables = new CalamariVariables();
+            variables.Set(variableName, variableValue);
+
+            var wrapper = CreateManifestReportScriptWrapper(variables);
+
+            wrapper.IsEnabled(Syntax).Should().BeTrue();
+            manifestReporter.DidNotReceive().ReportManifestApplied(Arg.Any<string>());
```
[the remark this answers is not in the document]
Comment (kevjt) [reply source/Calamari.Tests/KubernetesFixtures/ManifestReportScriptWrapperTests.cs:56]: Fair enough. I will remove the unnecessary assertion

File: source/Calamari.Tests/KubernetesFixtures/ManifestReportScriptWrapperTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (eddymoulton) [reply source/Calamari.Tests/KubernetesFixtures/ManifestReportScriptWrapperTests.cs:48]: Cool, I'm happy with it being consistent

File: source/Calamari.Tests/KubernetesFixtures/ManifestReportScriptWrapperTests.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+            var wrapper = CreateManifestReportScriptWrapper(variables, fileSystem);
+
+            wrapper.ExecuteScript(
+                                  new Script("stub"),
+                                  Syntax,
+                                  new CommandLineRunner(new SilentLog(), variables),
+                                  new Dictionary<string, string>());
+
+            foreach (var expectedManifest in GetExpectedManifests(testDirectory))
```
[the remark this answers is not in the document]
Comment (kevjt) [reply source/Calamari.Tests/KubernetesFixtures/ManifestReportScriptWrapperTests.cs:77]: Good call 👍

File: source/Calamari/Kubernetes/ManifestRetriever.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        }
+
+        IEnumerable<string> GetManifestFileNames(string workingDirectory)
+        {
+            var customResourceFileName =
+                variables.Get(SpecialVariables.CustomResourceYamlFileName) ?? "customresource.yml";
+
+            return new[]
+            {
+                "secret.yml", customResourceFileName, "deployment.yml", "service.yml", "ingress.yml",
```
[the remark this answers is not in the document]
Comment (kevjt) [reply source/Calamari/Kubernetes/ManifestRetriever.cs:48]: Yes, they will need to be added after https://github.com/OctopusDeploy/Calamari/pull/1401 is merged

**Review by eddymoulton — APPROVED** [review]

**Pull request merged**